File: api/market-data.js
// Vercel Serverless Function - 市场数据API
export default async function handler(req, res) {
  // 设置CORS头
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'GET, POST, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

  if (req.method === 'OPTIONS') {
    res.status(200).end();
    return;
  }

  try {
    const { symbol } = req.query;

    if (!symbol) {
      return res.status(400).json({
        success: false,
        message: '缺少symbol参数'
      });
    }

    // 模拟市场数据
    const mockData = {
      symbol: symbol.toUpperCase(),
      price: Math.random() * 50000 + 30000, // 30000-80000之间的随机价格
      change24h: (Math.random() - 0.5) * 10, // -5% 到 +5%
      volume24h: Math.random() * 1000000000, // 随机成交量
      high24h: Math.random() * 55000 + 35000,
      low24h: Math.random() * 45000 + 25000,
      openInterest: Math.random() * 500000000,
      fundingRate: (Math.random() - 0.5) * 0.001,
      lastUpdated: new Date().toISOString(),
      contractType: 'perpetual',
      exchange: 'Binance'
    };

    res.status(200).json({
      success: true,
      data: mockData
    });

  } catch (error) {
    console.error('Market data API error:', error);
    res.status(500).json({
      success: false,
      message: '服务器内部错误'
    });
  }
}

File: api/market-data.py
"""
Vercel Serverless Function - 市场数据API (Python版本)
"""

from http.server import BaseHTTPRequestHandler
import json
import urllib.parse
import random
import time
from datetime import datetime

class handler(BaseHTTPRequestHandler):
    def do_GET(self):
        # 设置CORS头
        self.send_response(200)
        self.send_header('Content-type', 'application/json')
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'GET, POST, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')
        self.end_headers()

        try:
            # 解析URL参数
            parsed_url = urllib.parse.urlparse(self.path)
            query_params = urllib.parse.parse_qs(parsed_url.query)
            
            symbol = query_params.get('symbol', [None])[0]
            
            if not symbol:
                response = {
                    'success': False,
                    'message': '缺少symbol参数'
                }
                self.wfile.write(json.dumps(response, ensure_ascii=False).encode('utf-8'))
                return

            # 生成模拟市场数据
            mock_data = {
                'symbol': symbol.upper(),
                'price': round(random.uniform(30000, 80000), 2),
                'change24h': round((random.random() - 0.5) * 10, 2),
                'volume24h': round(random.uniform(100000000, 1000000000), 2),
                'high24h': round(random.uniform(35000, 85000), 2),
                'low24h': round(random.uniform(25000, 75000), 2),
                'openInterest': round(random.uniform(100000000, 500000000), 2),
                'fundingRate': round((random.random() - 0.5) * 0.001, 6),
                'lastUpdated': datetime.now().isoformat(),
                'contractType': 'perpetual',
                'exchange': 'Binance'
            }

            response = {
                'success': True,
                'data': mock_data
            }

            self.wfile.write(json.dumps(response, ensure_ascii=False).encode('utf-8'))

        except Exception as e:
            response = {
                'success': False,
                'message': f'服务器内部错误: {str(e)}'
            }
            self.wfile.write(json.dumps(response, ensure_ascii=False).encode('utf-8'))

    def do_OPTIONS(self):
        # 处理预检请求
        self.send_response(200)
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'GET, POST, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')
        self.end_headers()
